.circleci/config.yml

```diff
@@ -0,0 +1,127 @@
+# Use the latest 2.1 version of CircleCI pipeline processing engine, see https://circleci.com/docs/2.0/configuration-reference/
+version: 2.1
+executors:
+  php-72:
+    docker:
+      - image: thecodingmachine/php:7.2-v2-cli
+jobs:
+  install:
+    executor: php-72
+    steps:
+      - checkout
+      - run: date "+%F" > /tmp/cachekey; cat composer.json >> /tmp/cachekey
```

Comment:
For non library PHP projects where composer.lock is committed that is normally used as a cache-key. Since composer.lock isn't committed here, and circleci caches never expire, I've used a combination of the date and the contents of the composer.json as the cache key. So when an upstream releases a new version that we can use ever build within the same calendar day will show the change. After that the new version of vendor will be cached.